generators/messaging-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/MessageListenerCodeGenerator.java

```diff
@@ -111,21 +118,32 @@ TypeSpec generatedCodeFor(final Resource resource, final BaseUri baseUri, final
                             .build())
                     .addAnnotation(messageDrivenAnnotation(component, resource.getActions(), resourceUri, baseUri, listenToAllMessages));
 
-            addLoggerAndSchemaInterceptorsIfNotGeneralJsonMimeType(resource, typeSpecBuilder);
+            if (!containsGeneralJsonMimeType(resource.getActions())) {
+                typeSpecBuilder.addAnnotation(AnnotationSpec.builder(Interceptors.class)
+                        .addMember(DEFAULT_ANNOTATION_PARAMETER, "$T.class", JmsLoggerMetadataInterceptor.class)
+                        .addMember(DEFAULT_ANNOTATION_PARAMETER, "$T.class", JsonSchemaValidationInterceptor.class)
+                        .build());
+            }
+
+            if (shouldAddCustomPoolConfiguration(generatorConfiguration)) {
+                typeSpecBuilder.addAnnotation(AnnotationSpec.builder(Pool.class)
+                        .addMember(DEFAULT_ANNOTATION_PARAMETER, "$S", poolNameFrom(resourceUri, component))
+                        .build());
+            }
 
             return typeSpecBuilder;
         }
 
         throw new IllegalStateException(format("JMS Endpoint generation is unsupported for framework component type %s", component));
     }
 
-    private void addLoggerAndSchemaInterceptorsIfNotGeneralJsonMimeType(final Resource resource, final TypeSpec.Builder typeSpecBuilder) {
```

Comment:
I just felt that the method name was a bit long and complex and that the code gained on readability when actually inlining the if statement.